File: frontend/src/types/Exam.ts
export type ExamStatus = 'DRAFT' | 'PUBLISHED' | 'COMPLETED';

export interface Exam {
  id: number;
  title: string;
  description: string;
  durationMinutes: number;
  passScore: number;
  status: ExamStatus;
  createdById?: number;
  createdByName?: string;
  totalQuestions?: number;
  startTime?: string;
  endTime?: string;
  assignedStudentIds?: number[];
  assignedStudentNames?: string[];
  assignedClassroomIds?: number[];
  assignedClassroomNames?: string[];
  createdAt?: string;
}

export interface CreateExamDto {
  title: string;
  description: string;
  durationMinutes: number;
  passScore: number;
  status?: ExamStatus;
  startTime?: string;
  endTime?: string;
  assignedStudentIds?: number[];
  assignedClassroomIds?: number[];
}

export interface UpdateExamDto {
  title?: string;
  description?: string;
  durationMinutes?: number;
  passScore?: number;
  status?: ExamStatus;
  startTime?: string;
  endTime?: string;
  assignedStudentIds?: number[];
  assignedClassroomIds?: number[];
}

File: backend/src/main/java/com/examplatform/exam/Exam.java
package com.examplatform.exam;

import com.examplatform.user.User;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "exams")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Exam {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "duration_minutes", nullable = false)
    @Builder.Default
    private Integer durationMinutes = 45;

    @Column(name = "pass_score", nullable = false)
    @Builder.Default
    private Double passScore = 5.0;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ExamStatus status = ExamStatus.DRAFT;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "exam_assigned_students",
        joinColumns = @JoinColumn(name = "exam_id"),
        inverseJoinColumns = @JoinColumn(name = "student_id")
    )
    @Builder.Default
    private java.util.Set<User> assignedStudents = new java.util.HashSet<>();

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "exam_assigned_classes",
        joinColumns = @JoinColumn(name = "exam_id"),
        inverseJoinColumns = @JoinColumn(name = "classroom_id")
    )
    @Builder.Default
    private java.util.Set<com.examplatform.classroom.Classroom> assignedClasses = new java.util.HashSet<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
